plugins/core/jetbrains-community/src/migration/software/aws/toolkits/jetbrains/core/coroutines/PluginCoroutineScopeTracker.kt

```diff
@@ -27,11 +31,28 @@ class PluginCoroutineScopeTracker : Disposable {
 }
 
 private class BackgroundThreadPoolScope(coroutineName: String, disposable: Disposable) : CoroutineScope {
-    override val coroutineContext = SupervisorJob() + CoroutineName(coroutineName) + getCoroutineBgContext()
+    override val coroutineContext = SupervisorJob() +
+        CoroutineName(coroutineName) +
+        PluginResolverDispatcher(getCoroutineBgContext() as ExecutorCoroutineDispatcher, PluginResolver.fromCurrentThread())
```

Comment:
not really sure if this is the right pattern or if we should be using context elements
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/as-context-element.html

Reply:
Good call. Updated approach uses a ThreadContextElement.